ext/liquid_c/tokenizer.c

```diff
@@ -114,6 +115,16 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
     token->length = cursor - tokenizer->cursor;
     tokenizer->cursor += token->length;
     tokenizer->length -= token->length;
+
+    if (tokenizer->line_number) {
+        const char *cursor = token->str;
+        const char *end = token->str + token->length;
+        while (cursor < end) {
+            if (*cursor == '\n')
+                tokenizer->line_number++;
+            cursor++;
+        }
+    }
```

Comment:
I would prefer if we could do this while iterating the first time, but I also realize the code would be a little messy and prone to missing cases :(